django/contrib/contenttypes/fields.py

```diff
@@ -163,20 +165,45 @@ def _check_content_type_field(self):
     def get_cache_name(self):
         return self.name
 
-    def get_content_type(self, obj=None, id=None, using=None):
+    def get_content_type(self, obj=None, id=None, using=None, model=None):
         if obj is not None:
             return ContentType.objects.db_manager(obj._state.db).get_for_model(
                 obj, for_concrete_model=self.for_concrete_model
             )
         elif id is not None:
             return ContentType.objects.db_manager(using).get_for_id(id)
+        elif model is not None:
+            return ContentType.objects.db_manager(using).get_for_model(
+                model, for_concrete_model=self.for_concrete_model
+            )
         else:
             # This should never happen. I love comments like this, don't you?
             raise Exception("Impossible arguments to GFK.get_content_type!")
 
     def get_prefetch_queryset(self, instances, queryset=None):
-        if queryset is not None:
-            raise ValueError("Custom queryset can't be used for this lookup.")
+        warnings.warn(
+            "Calling get_prefetch_queryset() is deprecated. "
+            "Use get_prefetch_querysets() instead.",
+            RemovedInDjango60Warning,
+        )
```

Comment:
This deprecation should be documented and tested, check out [docs](https://docs.djangoproject.com/en/4.2/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature).

Reply:
> should I add a section on the documentation about it?

No.

> In the release notes maybe?

Deprecation should be mentioned in release notes `5.0.txt` and `docs/internals/deprecation.txt`.

Reply:
Thank you! (I just noticed the what you said was written at the end of the deprecation doc, sorry)

Reply:
This warning is still untested.